src/server.rs

```diff
@@ -108,6 +160,20 @@ async fn websocket_handler(
         }
     };
 
+    // Record API key usage with a label for tracking
+    let key_value = match api_key.clone() {
+        Some(key) => {
+            // For security, only use the first 8 chars of the API key in metrics
+            if key.len() > 8 {
+                format!("{}...", &key[0..8])
+            } else {
+                key
+            }
+        }
+        None => "none".to_string(),
+    };
+    counter!("websocket_proxy.connections_by_api_key", "key" => key_value).increment(1);
```

Comment:
Instead of this I wonder if we should take in api keys in the format:

<application>:<key>,<application2>:<key2>

Then we can log out the application info, it'll make it much easier to debug issues that way too (and no need to partially leak the key)